File: answers/attempt_2/python/utility/binary_to_number.py
def convert_binary_to_number( binary ):
  converted_number = 0
  power = len( binary ) - 1

  for bin in binary:
    if bin == '1':
      power_of_m = 2 ** power
      converted_number += power_of_m

    power -= 1
  
  return converted_number

File: answers/attempt_1/javascript/utilityfunctions/binarytonumber.js
module.exports = function convertBinaryToNumber( binary ) {
  let binaryArrayed = binary.split('');

  let convertedBinaryToNumber = 0;

  let power = binary.length - 1;

  for( let i=0; i<binary.length; i++ ) {
    if( binaryArrayed[i] === '1' ) {
      let twoPower = Math.pow( 2, power );

      convertedBinaryToNumber += twoPower;
    }

    power -= 1;
  }

  return convertedBinaryToNumber;
};